### PR TITLE
Add Update Contract

### PR DESCRIPTION
**Pull request by SethDusek**
Hi, I've added the update contract for #51

I had some questions regarding the PR though
- What is the best way to obtain the index for a certain constant? Right now the way I used is to just print all the constants, and see which one matches the constant in the source code for update contract. 
- Is submitting to chain/unit testing similar to core/pool_commands/refresh.rs part of this issue?

### CODE REVIEW

**Comment by greenhat** [comment]
> Hi, I've added the update contract for #51

Great! Thank you!

> 
> I had some questions regarding the PR though
> 
>     * What is the best way to obtain the index for a certain constant? Right now the way I used is to just print all the constants, and see which one matches the constant in the source code for update contract.

I'm using the same technique and I'm unaware of a better way to do it. There is an EIP for named constant support for the compiler but it's not implemented.

> 
>     * Is submitting to chain/unit testing similar to core/pool_commands/refresh.rs part of this issue?

I'm not sure I understand the question. The submission (update command after voting) should be implemented in #49. This (#51) issue is just for the contract parametrization in the `contracts` module.

**Comment by SethDusek** [comment]
> I'm using the same technique and I'm unaware of a better way to do it. There is an EIP for named constant support for the compiler but it's not implemented.

Hmm, if ergoscript-compiler (from sigma-rust) is mature enough maybe that could be used to compile it. That way we could also have the source code for contracts part of the contract itself instead of obtaining the compiled version from Playground. If there's some way to inspect the AST we could maybe use that to obtain constant indexes too, but I'm not too familiar with ergoscript-compiler's inner workings

**Comment by greenhat** [comment]
> > I'm using the same technique and I'm unaware of a better way to do it. There is an EIP for named constant support for the compiler but it's not implemented.
> 
> Hmm, if ergoscript-compiler (from sigma-rust) is mature enough maybe that could be used to compile it. That way we could also have the source code for contracts be part of the UpdateContract/PoolContract/etc itself instead of obtaining the compiled version from Playground. 

ergoscript-compiler is roughly 0.5% ready, so no, but it'd be great to have the contract's ErgoScript code along with the off-chain code.

> If there's some way to inspect the AST we could maybe use that to obtain constant indexes too, but I'm not too familiar with ergoscript-compiler's inner workings

Constant indices are built during the serialization of an AST where constants are swapped with placeholders with an index. I cannot think of any AST analysis that would help here.

Here is the EIP-5 I mentioned earlier - https://github.com/ergoplatform/eips/blob/master/eip-0005.md . Unfortunately, it's not been implemented yet.

**Comment by SethDusek** [comment]
Yup, it's good to go

File: core/src/contracts/update.rs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
...dCA9IHVwZGF0ZUJveE91dC50b2tlbnMgPT0gU0VMRi50b2tlbnMgICAgICAgICAgICAgICAgICAgICAmJgogICAgICAgICAgICAgICAgICAgICAgIHVwZGF0ZUJveE91dC5wcm9wb3NpdGlvbkJ5dGVzID09IFNFTEYucHJvcG9zaXRpb25CeXRlcyAmJgogICAgICAgICAgICAgICAgICAgICAgIHVwZGF0ZUJveE91dC52YWx1ZSA+PSBTRUxGLnZhbHVlICAgICAgICAgICAgICAgICAgICAgICAmJgogICAgICAgICAgICAgICAgICAgICAgIHVwZGF0ZUJveE91dC5jcmVhdGlvbkluZm8uXzEgPiBTRUxGLmNyZWF0aW9uSW5mby5fMSAgICAmJgogICAgICAgICAgICAgICAgICAgICAgICEgKHVwZGF0ZUJveE91dC5SNFtBbnldLmlzRGVmaW5lZCkgCgogIGRlZiBpc1ZhbGlkQmFsbG90KGI6Qm94KSA9IGlmIChiLnRva2Vucy5zaXplID4gMCkgewogICAgYi50b2tlbnMoMCkuXzEgPT0gYmFsbG90VG9rZW5JZCAgICAgICAmJgogICAgYi5SNVtJbnRdLmdldCA9PSBTRUxGLmNyZWF0aW9uSW5mby5fMSAmJiAvLyBlbnN1cmUgdm90ZSBjb3JyZXNwb25kcyB0byB0aGlzIGJveCBieSBjaGVja2luZyBjcmVhdGlvbiBoZWlnaHQKICAgIGIuUjZbQ29sbFtCeXRlXV0uZ2V0ID09IHBvb2xPdXRIYXNoICAgICAgLy8gY2hlY2sgdmFsdWUgdm90ZWQgZm9yCiAgfSBlbHNlIGZhbHNlCiAgCiAgdmFsIGJhbGxvdEJveGVzID0gSU5QVVRTLmZpbHRlcihpc1ZhbGlkQmFsbG90KQogIAogIHZhbCB2b3Rlc0NvdW50ID0gYmFsbG90Qm94ZXMuZm9sZCgwTCwgeyhhY2N1bTogTG9uZywgYjogQm94KSA9PiBhY2N1bSArIGIudG9rZW5zKDApLl8yfSkKICAKICBzaWdtYVByb3AodmFsaWRQb29sSW4gJiYgdmFsaWRQb29sT3V0ICYmIHZhbGlkVXBkYXRlT3V0ICYmIHZvdGVzQ291bnQgPj0gbWluVm90ZXMpICAKfQ==
```
Comment (greenhat) [inline core/src/contracts/update.rs:26]: There is an updated v2.0a version of the update box contract at https://github.com/scalahub/OraclePool/blob/58726e7751b2ff454ce4b3f45315f3cd549c2994/src/main/scala/oraclepool/v2a/Contracts.scala#L213-L270

**Review by greenhat — CHANGES_REQUESTED** [review]
Looking good! Please, use the updated v2.0a version of the contract (see my comment).

**Pull request merged**